### PR TITLE
zh,en: make the description clearer on how to use the configs

### PR DESCRIPTION
**Pull request by Liuxiaozhen12**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-dm) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)

Make the description clearer to close https://github.com/pingcap/docs-dm/issues/733.
### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs-dm/issues/733

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/task-configuration-guide.md [hunk 1]
```diff
@@ -162,7 +162,8 @@ target-database:       # 目标 TiDB 配置
 
 ## ********* 任务信息配置 *********
 name: test                      # 任务名称，需要全局唯一
-shard-mode: "pessimistic"       # 如果为分库分表合并任务则需要配置该项。默认使用悲观协调模式 "pessimistic"，在深入了解乐观协调模式的原理和使用限制后，也可以设置为乐观协调模式 "optimistic"
+is-sharding: true
```
Comment (Liuxiaozhen12) [inline zh/task-configuration-guide.md:165]: I'm not sure if we should add the config item like this (as mentioned in the related issue). If we do, should I add it in all the DM task configuration file? Please help to review it, thx! @lance6716

File: zh/task-configuration-file-full.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ aliases: ['/docs-cn/tidb-data-migration/dev/task-configuration-file-full/','/zh/
 ## ********* 基本信息配置 *********
 name: test                      # 任务名称，需要全局唯一
 task-mode: all                  # 任务模式，可设为 "full" - "只进行全量数据迁移"、"incremental" - "Binlog 实时同步"、"all" - "全量 + Binlog 实时同步"
-shard-mode: "pessimistic"       # 任务协调模式，可选的模式有 ""、"pessimistic、"optimistic"。默认使用 "" 即无需协调。如果是分库分表合并任务，请设置为悲观协调模式 "pessimistic"。
+shard-mode: "pessimistic"       # 任务协调模式，可选的模式有 ""、"pessimistic、"optimistic"。默认值为 "" 即无需协调。如果是分库分表合并任务，请设置为悲观协调模式 "pessimistic"。
```
Comment (hfxsd) [inline zh/task-configuration-file-full.md:27]: 这里需要补上这句话吗？ 
在深入了解乐观协调模式的原理和使用限制后，也可以设置为乐观协调模式 "optimistic"

**Review by hfxsd — APPROVED** [review]
LGTM

File: zh/task-configuration-guide.md (hunk 1 above)
Comment (glorv) [reply zh/task-configuration-guide.md:165]: `is-sharding` is a deprecated config field and its meaning is cover by `shard-mode`.   So we need explicitly set it. This field is only used for the compatibility of legacy configs.

**Review by glorv — COMMENTED** [review]
rest LGTM

File: zh/task-configuration-file-full.md (hunk 2 above)
Comment (Liuxiaozhen12) [reply zh/task-configuration-file-full.md:27]: 下面有的，就不用补了

File: zh/key-features.md [hunk 3]
```diff
@@ -446,10 +446,10 @@ DM 支持将上游 MySQL/MariaDB 各分库分表中的 DML、DDL 数据合并后
 
 ### 参数配置
 
-在 task 的配置文件中设置：
+在 task 的配置文件中将 `is-sharding` 设置为 `true`：
```
Comment (glorv) [inline zh/key-features.md:449]: `is-sharding` 不推荐再显示申明了，建议新的示例配置里面都不要提这个字段

File: en/feature-online-ddl.md [hunk 4]
```diff
@@ -52,7 +52,7 @@ In the task configuration file, `online-ddl` is at the same level of `name`. For
 ## ********* Basic configuration *********
 name: test                      # The name of the task. Should be globally unique.
 task-mode: all                  # The task mode. Can be set to `full`/`incremental`/`all`.
-shard-mode: "pessimistic"       # You need to configure this if it is a shard merge task. The "pessimistic" mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set it to the "optimistic" mode.
+shard-mode: "pessimistic"       # The shard merge mode. Optional modes are ""/"pessimistic"/"optimistic". The "" mode is used by default which means no specific mode is required. If the task is a shard merge task, set it to the "pessimistic" mode. After understanding the principles and restrictions of the "optimistic" mode, you can set it to the "optimistic" mode.
```
Comment (glorv) [inline en/feature-online-ddl.md:55]:
```suggestion
shard-mode: "pessimistic"       # The shard merge mode. Optional modes are ""/"pessimistic"/"optimistic". The "" mode is used by default which means sharding DDL merge is disabled. If the task is a shard merge task, set it to the "pessimistic" mode. After understanding the principles and restrictions of the "optimistic" mode, you can set it to the "optimistic" mode.
```

File: zh/faq.md [hunk 5]
```diff
@@ -145,7 +145,7 @@ DM 在最后 `rename ghost_table to origin table` 的步骤会把内存的 DDL 
 
 ## DM v2.0.0 启动任务时出现 `fail to initial unit Sync of subtask`，报错信息的 `RawCause` 显示 `context deadline exceeded`
 
-该问题是 DM v2.0.0 版本的已知问题，在同步任务的表数目较多时触发，将在 v2.0.1 修复。使用 TiUP 部署的用户可以升级到开发版 nightly 解决该问题，或者访问 GitHub 上 [DM 仓库的 release 页面](https://github.com/pingcap/ticdc/releases)下载 v2.0.0-hotfix 版本手动替换可执行文件。
+该问题是 DM v2.0.0 的已知问题，在同步任务的表数目较多时触发，将在 v2.0.1 修复。使用 TiUP 部署的用户可以升级到开发版 nightly 解决该问题，或者访问 GitHub 上 [DM 仓库的 release 页面](https://github.com/pingcap/dm/releases)下载 v2.0.0-hotfix 版本手动替换可执行文件。
```
Comment (glorv) [inline zh/faq.md:148]: dm 合并到 cdc 之后，最新的 release 也会在 cdc 仓库，我们这里是不是还是放 cdc release 的链接比较合理，用户看到的就是最新版本的 release 了。(v2.x 的 release 将不会包含在里面)

**Review by glorv — COMMENTED** [review]
rest LGTM

File: zh/faq.md (hunk 5 above)
Comment (Liuxiaozhen12) [reply zh/faq.md:148]: 喔喔谢谢提醒！这里解决冲突的时候搞错了

Comment (Liuxiaozhen12) [inline zh/faq.md:148]:
```suggestion
该问题是 DM v2.0.0 的已知问题，在同步任务的表数目较多时触发，将在 v2.0.1 修复。使用 TiUP 部署的用户可以升级到开发版 nightly 解决该问题，或者访问 GitHub 上 [DM 仓库的 release 页面](https://github.com/pingcap/ticdc/releases)下载 v2.0.0-hotfix 版本手动替换可执行文件。
```

**Pull request merged**